cmd/tendermint/commands/key_migrate.go

```diff
@@ -6,70 +6,71 @@ import (
 
 	"github.com/spf13/cobra"
 
-	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/config"
 	"github.com/tendermint/tendermint/libs/log"
 	"github.com/tendermint/tendermint/scripts/keymigrate"
 	"github.com/tendermint/tendermint/scripts/scmigrate"
 )
 
-func MakeKeyMigrateCommand(conf *cfg.Config, logger log.Logger) *cobra.Command {
+func MakeKeyMigrateCommand(conf *config.Config, logger log.Logger) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "key-migrate",
 		Short: "Run Database key migration",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx, cancel := context.WithCancel(cmd.Context())
-			defer cancel()
+			return RunDatabaseMigration(cmd.Context(), logger, conf)
+		},
+	}
 
-			contexts := []string{
-				// this is ordered to put
-				// the more ephemeral tables first to
-				// forclose the possiblity of the
-				// ephemeral data overwriting later data
-				"tx_index",
-				"peerstore",
-				"light",
-				"blockstore",
-				"state",
-				"evidence",
-			}
+	// allow database info to be overridden via cli
+	addDBFlags(cmd, conf)
 
-			for idx, dbctx := range contexts {
-				logger.Info("beginning a key migration",
-					"dbctx", dbctx,
-					"num", idx+1,
-					"total", len(contexts),
-				)
+	return cmd
+}
 
-				db, err := cfg.DefaultDBProvider(&cfg.DBContext{
-					ID:     dbctx,
-					Config: conf,
-				})
+func RunDatabaseMigration(ctx context.Context, logger log.Logger, conf *config.Config) error {
+	contexts := []string{
+		// this is ordered to put
+		// the more ephemeral tables first to
+		// forclose the possiblity of the
+		// ephemeral data overwriting later data
+		"tx_index",
+		"peerstore",
+		"light",
+		"blockstore",
+		"state",
+		"evidence",
+	}
 
-				if err != nil {
-					return fmt.Errorf("constructing database handle: %w", err)
-				}
+	for idx, dbctx := range contexts {
+		logger.Info("beginning a key migration",
+			"dbctx", dbctx,
+			"num", idx+1,
+			"total", len(contexts),
+		)
```

Comment:
Can dbctx include the full path of the store

Reply:
we could munge together the filename ourselves, based on the inputs to the function, but the path of the store is determined by the behavior of the db provider.

Reply:
The reason I ask is just so that users know that it is pointed to the correct db

Reply:
the dbctx value you will tell you which db file we’re pointed at, which I think is a big help, if the path is wrong, we’d not see that. could add another log message to the beginning to say where the dbpath is